### PR TITLE
Add workflow to publish docker image for self-hosting

### PR DESCRIPTION
**Pull request by alexklibisz**
Re: #26 

Pretty straightforward: checkout the latest tag, build the docker image for the Python flask application, and publish it to Github's container registry, using the git tag as the image tag.

Here's how it looks when it runs: https://github.com/alexklibisz/explain.dalibo.com/actions/runs/8943147452

Here's how it looks when the image is published: https://github.com/alexklibisz/explain.dalibo.com/pkgs/container/pev2/versions

We might want to adjust the name. I chose `pev2` for now, but perhaps you want to call it something else.

### CODE REVIEW

File: .github/workflows/publish-docker.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: ${{ github.ref == 'refs/heads/main' }}
+          platforms: linux/amd64,linux/arm64
+          tags: |
+            ghcr.io/${{ github.repository_owner }}/pev2:${{ env.TAG }}
+            ghcr.io/${{ github.repository_owner }}/pev2:latest
```
Comment (pushkar-anand) [inline .github/workflows/publish-docker.yaml:40]: Hey @alexklibisz I was trying to self-host the service and found this PR. I forked the repo and tried using the workflow to build the image for my usage but found an issue.

The tag is of the format `[HOST[:PORT_NUMBER]/]PATH` where the path is `[NAMESPACE/]REPOSITORY`.

In this case, the tag: `ghcr.io/dalibo/pev2:<tag>` is invalid. I tried to change it to `ghcr.io/dalibo/explain.dalibo.com/pev2:<tag>` which is also invalid.

See: https://github.com/docker/setup-buildx-action/discussions/291 & https://docs.docker.com/reference/cli/docker/image/tag/

Comment (pushkar-anand) [reply .github/workflows/publish-docker.yaml:40]: This is what I ended up using to make it work.

```suggestion
            ghcr.io/${{ github.repository_owner }}/explain-dalibo-com/pev2:${{ env.TAG }}
            ghcr.io/${{ github.repository_owner }}/explain-dalibo-com/pev2:latest
```

**Comment by alexklibisz** [comment]
I see some comments from @pushkar-anand in my email, but not on the PR:
image

**Comment by alexklibisz** [comment]
@pushkar-anand if you want to self-host the service, I have an image hosted here: https://github.com/alexklibisz/pev2-docker/pkgs/container/pev2-server

**Comment by pushkar-anand** [comment]
> I see some comments from @pushkar-anand in my email, but not on the PR:
> image
> 

Yes, I got it working so I deleted it. My bad

**Comment by alexklibisz** [comment]
@pgiraud Could you have a look here? Thanks

**Comment by alexklibisz** [comment]
No problem at all. Thanks for picking it up! I'll go ahead and close here since this is covered in #33.

**Pull request closed without merging**